noir/noir-repo/aztec_macros/src/transforms/storage.rs

```diff
@@ -214,48 +214,22 @@ fn get_serialized_length(
             secondary_message: Some("State storage variable must be generic".to_string()),
         })?;
 
-    let is_note = traits.iter().any(|&trait_id| {
-        let r#trait = interner.get_trait(trait_id);
-        r#trait.name.0.contents == "NoteInterface"
-            && !interner.lookup_all_trait_implementations(stored_in_state, trait_id).is_empty()
-    });
+    let is_note = match stored_in_state {
+        Type::Struct(typ, _) => interner
+            .struct_attributes(&typ.borrow().id)
+            .iter()
+            .any(|attr| is_custom_attribute(attr, "aztec(note)")),
+        _ => false,
+    };
 
     // Maps and (private) Notes always occupy a single slot. Someone could store a Note in PublicMutable for whatever reason though.
     if struct_name == "Map" || (is_note && struct_name != "PublicMutable") {
         return Ok(1);
```

Comment:
Not related to this PR, but this assumes `PublicMutable` stores field-like values. I could very well store a struct in one. See #5736.

Reply:
Mmm, it doesn't? This is just to short-circuit looking for the serialized size in case we find a map ("occupies" a single slot) or we find a note stored in a Private wrapper (also a single slot)

Reply:
I misread the if condition :facepalm: